# Een modelletje van economische ongelijkheid

Auteur: [Dirk Roorda](https://pure.knaw.nl/portal/en/persons/dirk-roorda)

Datum: (eerste online verschijning: 2021-12-05)

Bron: dit werk staat op GitHub: [dirkroorda/yardsale](https://github.com/dirkroorda/yardsale).

## Inleiding

Dit notebook is gebaseerd op dit
[artikel in de Scientific American](https://www.scientificamerican.com/article/is-inequality-inevitable/) en op deze
[simulatie van het Yard Sale model](http://www.physics.umd.edu/hep/drew/math_general/yard_sale.html),
dat in het artikel opgevoerd wordt.

Het artikel beweert dat economisch verkeer aan wetmatigheden onderworpen kan zijn die
op kleine schaal bezien eerlijk kunnen lijken, maar op grote schaal
kunnen leiden tot extreme ongelijkheid.

In het Yard Sale model doen actoren zaken met elkaar, en bij elke transactie verliezen ze wat of
winnen ze wat, met 50% kans dat ze winnen of verliezen.
Het bedrag dat ze winnen of verliezen is een percentage van het vermogen van de armste speler in de transactie. Een gebruikelijke instelling is:

* bij winst krijgt de armste speler 20% van zijn vermogen erbij;
* bij verlies raakt de armste speler 17% van zijn vermogen kwijt.

### De les van het Yard Sale model

Bij elke individuele transactie zit de armere partij in een situatie: ik kan een groter bedrag
winnen of een kleiner bedrag verliezen, dus, waarom zou ik het niet doen?

Maar in een simulatie van het spel zie je dat bijna alle spelers alles gaan verliezen,
terwijl één speler of een handjevol al het kapitaal gaat vergaren.

De moraal is: een systeem dat op kleine schaal redelijk eerlijk en zelfs voordelig lijkt voor
de armere partij, kan op grote schaal zorgen voor extreme ongelijkheid.

### Het Yard Sale model te licht bevonden

Op de
[Open Discussie Najaar 2021 van KlimaatVeranda](https://klimaatveranda.nl/2021/10/22/open-discussie-najaar-2021/#comments) is een gesprek gevoerd over de rol van de economische wetenschap bij
het oplossen van het klimaatprobleem. Daarbij heb ik gewezen op het Yard Sale model met als les
dat er een natuurlijke tendens bestaat die ongelijkheid bevordert, een tendens die wiskundig
van aard is.

Vervolgens vroegen de gesprekspartners zich af of dit een terecht getrokken les was.
Ik beperk me hier even tot de inbreng van Hans Custers, die het model analyseerde en zag
dat het ongelijkheidseffect er door de parameterkeuze gewoon ingebakken zat.
Daarop concludeerde hij dat dit model geen waarde had in het beschrijven van de economische werkelijkheid.

### Te snel verworpen

Die analyse vond ik overtuigend, maar de conclusie niet. Terwijl volgens Hans het model triviaal
is als je zijn analyse volgt, bleef bij mij het gevoel knagen dat er toch niet alles mee gezegd was.

In wat volgt leg ik uit wat blijft knagen, en maak ik een simulatie van een variant van het model
die het gedrag ervan tot in detail laat zien.

Daaruit trek ik dan weer de conclusie dat dit model wèl interessante dingen kan zeggen over
de economische werkelijkheid.

## Analyse

Hier komt de essentie van de analyse van Hans:

Je kunt ook anders naar de winst en verliesbedragen kijken, niet als losse bedragen
die je wint of verliest, maar als *factoren* waarmee je vermogen groeit of krimpt.


Dan ziet het er anders uit:

* factor bij winst: 1 + 0,2 = 1,2
* factor bij verlies: 1 - 0,17 = 0,83

Heft dat elkaar op?

```
1,2 * 0,83 = 0,996
```

Nee, de armere partij zal, als hij steeds tegen een rijkere partij speelt,
zijn kapitaal onherroepelijk zien krimpen.
Immers, hij zal ongeveer even vaak winnen als verliezen,
dus de verwachte groei factor op langere termijn is 0,996.

Als je de factoren iets anders kiest, namelijk 20 en 16, dan zie je in de simulatie
*niet* het extreme ongelijkheidseffect!

Immers

```
1,2 * 0,84 = 1,008
```

De verwachte groeifactor is nu 1,008.
Dit maakt dat er geen tendens is om de arme armer te maken.

## Gevolg van de analyse

Hier geef ik de strekking weer van de gevolgtrekking die Hans uit zijn analyse trekt.

*De ongelijkheid is door de parameterkeuze al in het systeem ingebakken.
Het zegt dan ook niets over de werkelijke wereld, maar alleen iets over dit wiskundige modelletje.
In de werkelijke wereld zullen mensen niet zo stom zijn om zo'n spel mee te spelen.
Daar heeft ongelijkheid andere oorzaken, van niet wiskundige aard.
Het is dan ook een staaltje van simplistisch denken als je meent 
de oorzaken van ongelijkheid uit de wiskunde te kunnen halen.*

## Een andere kijk

Terwijl ik de analyse deel, trek ik er andere conclusies uit.

Het zou zo maar kunnen zijn dat het model wel degelijk iets zegt over de werkelijke wereld.
Het feit dat wij aan de parameterknoppen kunnen draaien en daarmee een bepaald systeemgedrag
kunnen triggeren, betekent niet dat je in de werkelijke wereld aan dezelfde knoppen kunt draaien.

Je kunt het vriespunt van water uit een moleculair model afleiden waar de sterkte van ee interactie tussen de watermoleculen en de grootte van de watermoleculen als parameters in zitten.
Bij de goede keuze van de parameters rolt er dan het goede vriespunt van water uit.
Dat dat vriespunt puur het wiskundige gevolg van het model en zijn parameterkeuze is niet iets
om je druk over te maken, het is juist mooi.

Zo kan het best zo zijn dat het gedrag van de werkelijke economische wereld precies lijkt op dit model onder een specifieke keuze van parameters, waarbij die parameters meer aan de 20:17 kant liggen dan
aan de 20:16 kant.

Ten tweede, het zou zo maar kunnen zijn dat wij mensen in de werkelijke wereld inderdaad
dit soort spellen spelen, zonder dat we ons bewust zijn van het risico dat we daarmee lopen.
En dat komt omdat ogenschijnlijk de transacties in ons voordeel lijken.
Het gaat pas mis als die transacties (langdurig) herhaald worden.

Kortom, het zou kunnen dat we een stel regels op microschaal als eerlijk ervaren,
terwijl diezelfde regels op grote schaal en over lange tijd extreme ongelijkheid veroorzaken.

Bijvoorbeeld: het werkelijke systeem kan eerst een decennium in een regime verkeren
waar de ongelijkheden nog niet optreden, om vervolgens daarna de spelers
in een onafwendbare race-to-the-bottom te duwen.

Zit er iets in het Yard Sale model dat daar iets over zegt?

## Verwarrende intuïties

De analyse van Hans geeft een scherp beeld van het gedrag van het systeem op lange termijn,
het *asympotische* gedrag.

Ondertussen bleef ik zitten met bepaalde vragen over hoe het systeem in vredesnaam dat asymptotische
gedrag bereikt. Tijdens het voeren van de discussie op Klimaatveranda was ik daar nog niet uit
(nu wel). Bijvoorbeeld dit:

Hoe kan het dat een extreem rijke partij, die dus vrijwel altijd
met een armere partij zaken doet, daar beter van wordt?

Een rijkere partij die met iemand zaken doet die 100 euro heeft, zal ofwel 20 euro aan die
partij moeten geven, of hij krijgt er 17 euro van.
Als een *oligarch* (een extreem rijk geworden speler)
1000 transacties doet met allemaal verschillende armeren,
zal hij daar verlies op lijden. 
Want bij elk van die transacties is de kans dat hij
een iets kleiner bedrag krijgt van de ander even groot als
de kans dat hij een iets groter bedrag aan de ander moet betalen.

Maar als hij 1000 transacties doet met dezelfde armere partij, zal hij daar winst
op behalen, vanwege de krimpfactor uit de analyse van Hans.

Wat betekent dat voor het scenario waarbij de rijke partij een miljoen transacties doet met
1000 armere partijen? Welke systeemgedragingen kun je in dit scenario verwachten?

## De bank-variant van het spel

Om dit goed uit te zoeken, verzinnen we een iets ander spel.
We gaan uit van een situatie in het Yard Sale model waarbij 1 speler heel erg rijk geworden
is: dat is de bank geworden.
Vergeleken met de bank zijn alle andere spelers arm.

Het spel wordt in rondes gespeeld, en in elke ronde doet de bank met elke speler precies één transactie.
We zetten de winst en verlies bedragen beide op 20%. 
De winst en verlies factoren worden dus 1,2 en 0,8.
De krimpfactor is daarmee 1,2 * 0,8 = 0,96.
Dat is dus een forse krimpfactor, veel forser dan de 20:17 van het oorspronkelijke Yard Sale model.
Het spel heeft dus een nog sterkere tendens om alle spelers arm te maken ten gunste van de bank.

We maken het nog wat simpeler: de speler geldt altijd als de arme partij, zelfs als een speler toevallig
rijker is dan de bank.
Verder mag de bank rood staan.
Spelers zullen nooit rood staan, want ze kunnen nooit meer dan 20% van hun vermogen verliezen
bij elke transactie.
Ze kunnen wel vrijwel blut raken.

Samenvattend, de regels zijn:

* De bank heeft een groot kapitaal en speelt tegen veel spelers in veel rondes
* In elke ronde werpt iedere speler een munt en:
  * bij kop wint hij 20% van zijn huidige vermogen
  * bij munt verliest zij 20% van haar huidige vermogen
  * in beide gevallen wordt met de bank afgerekend
  * de bank mag rood staan

Merk op dat:

* de winstverwachting voor een speler die 1 ronde speelt is neutraal:
  geen winst en verlies;
* de winstverwachting voor veel spelers die elk 1 ronde spelen is neutraal:
  samen geen winst of verlies, maar de onderlinge verschillen tussen spelers
  kunnen iets toenemen;
* als het spelerskapitaal gelijk verdeeld is over de spelers, 
  is de winstverwachting voor de bank na 1 ronde neutraal;
* de winstverwachting voor een speler die veel rondes speelt is negatief:
  al zijn bezit zal naar de bank gaan.

## Het "vroege" gedrag van het systeem

We gaan nu het "vroege" gedrag van het systeem bestuderen,
namelijk wat het doet voordat de definitieve race-to-the-bottom voor de spelers
heeft ingezet.

Eerst nog een mogelijk verwarrende intuïtie.

Er is een redenering waarmee je kunt aantonen dat de bank gemiddeld quitte blijft spelen.
We zullen die redenering zometeen uitspellen.
Dan zal blijken dat die redenering alleen in het begin opgaat.
Pas daarna komt het asymptotische gedrag opzetten.

Wat blijkt: hoe meer spelers, hoe later het asympotische gedrag komt opzetten.

Hier is een tabel:

aantal deelnemers | aantal zetten vóór asymptotisch gedrag
--- | ---
2 | 1
4 | 2
8 | 3
64| 6
~1000 | 10
~1.000.000 | 20

Gedurende dit aantal zetten zal het systeem min of meer eerlijk aanvoelen.
In de werkelijke wereld zou een ronde gelijk kunnen staan aan een jaar.
En de bank zou een miljoen klanten kunnen hebben.
Dan duurt het dus 20 jaar voordat het duidelijk wordt dat de bank al het kapitaal
naar zich gaat toehalen.

Uit de simulaties die we hieronder zullen zien, blijkt dat de echte race-to-the-bottom
nog aanzienlijk later plaatsvindt.

## Vóór de race-to-the-bottom

Hier is hoe het gaat in de eerste zetten.

Stel dat we 1024 spelers hebben, elk met 125 euro.

**na ronde 1**

De verwachting is dat je 512 spelers hebt met 150 euro en 512 met 100 euro.
Dat betekent dat de bank nog evenveel geld heeft.

groep | aantal spelers | bedrag van elke speler in deze groep
--- | --- | ---
w | 512 | 150
v | 512 | 100

De spelers hebben in totaal nog net zoveel als voor de ronde, alleen anders verdeeld.

**na ronde 2**

Kijk per groep van hierboven. 
Elke groep splitst zich in tweeën: de winners en verliezers van ronde 2.
We krijgen vier groepen:

oude groep | nieuwe groep | aantal spelers | bedrag van elke speler in deze groep
---| --- | --- | ---
w | ww | 256 | 180
w | wv | 256 | 120
v | vw | 256 | 120
v | vv | 256 | 80

De spelers hebben in totaal nog net zoveel als voor de ronde, alleen anders verdeeld.

**na ronde 3**

oude groep | nieuwe group | aantal spelers | bedrag van elke speler in deze groep
--- | --- | --- | ---
ww | www | 128 | 216
ww | wwv | 128 | 144
wv | wvw | 128 | 144
wv | wvv | 128 | 96
vw | vww | 128 | 144
vw | vwv | 128 | 96
vv | vvw | 128 | 96
vv | vvv | 128 | 64

De spelers hebben in totaal nog net zoveel als voor de ronde, alleen anders verdeeld.

Dit gaat zo door totdat
uiteindelijk, na 10 rondes, er 1024 groepen zijn ontstaan van elk 1 persoon.

Daarna stopt de redenering. We kunnen de groepen niet verder onderverdelen.
De spelers worden minder en minder homogeen, en dat is precies de situatie waar de bank
winst uit kan halen.

## Spelverlopen

Dit effect kun je als volgt analyseren.

Als je 10 rondes speelt, heb je 1024 mogelijke spelverlopen per speler. Want 10 keer
achter elkaar kun je of winnen of verliezen en 2 ** 10 = 1024.

Met 1024 spelers heb je nog een redelijke kans dat alle mogelijke spelverlopen ook echt optreden.
Dus spelverlopen waar spelers heel vaak winnen en dus veel geld krijgen.
Daar staan evenzovele spelverlopen tegenover waar spelers heel vaak verliezen.

Maar wat de verliezers verliezen, is minder dan wat de winnaars winnen, omdat het winbedrag
exponentieel toeneemt met een positieve exponent, en het verliesbedrag met een negatieve exponent.

Neem als voorbeeld 2 spelers, W en V, elk 125 euro, 3 rondes, waarbij W drie keer wint en V drie keer
verliest.

speler | start | na ronde 1 | na ronde 2 | na ronde 3 | verschil
--- | --- | --- | --- | --- | ---
W | 125 | 150 | 180 | 216 | 91
V | 125 | 100 | 80 | 64 | 61

Dus de winnaar heeft 91 euro gewonnen, wat 30 euro meer is dan de 61 euro die de
verliezer verloren heeft.
Dit soort spelverlopen leiden dus tot voordeel van het spelerscollectief ten opzichte van de bank.

Maar wat gebeurt er als je veel rondes speelt? Het aantal mogelijke spelverlopen
neemt exponentieel toe. Voor 1000 rondes heb je wel ruwweg `(2 ** 10) ** 100` spelverlopen,
dat is een 1 met 300 nullen. Dat is meer dan astronomisch.
Er zullen maar een fractie van deze spelverlopen optreden.
En welke zijn dat? Die waarin het aantal keer verlies ongeveer even groot is als het
aantal keer winst.

En dat zijn precies de spelverlopen waarin de spelers verliezen, omdat daar de krimpfactor
van Hans zijn werk doet.

## Voorlopige conclusie

Zolang de bank met veel spelers zaken doet, en dat voor een beperkte tijd kan doen,
wordt de bank niet rijker ten koste van de spelers.
Maar de onderlinge ongelijkheid van de spelers neemt wel steeds toe.

## Onverwacht gedrag

De simulatie die we zometeen gaan volgen laat een (voor mij) onverwacht
gedrag zien, dat niet altijd optreedt, maar ook niet zeldzaam is.

Soms bereiken de spelers collectief overwicht over de bank, voor korte tijd,
om daarna gezamenlijk alsnog de afgrond in te gaan.
Soms gaat dat gepaard met nog een paar kleinere pieken.

Toen ik dat voor het eerst zag optreden dacht ik dat ik het fout geprogrammeerd had,
maar na wat inzoomen blijkt het een heel verklaarbaar verschijnsel te zijn.

Het is nu tijd om de handen maar eens vuil te maken. 
We gaan simuleren.

De code voor de simulatie zijn 100 regels Python, die je kunt nalezen in
[spel.py](https://github.com/dirkroorda/yardsale/blob/master/spel.py).

En die gebruikt de 50 regels Python van
[utils.py](https://github.com/dirkroorda/yardsale/blob/master/spel.py)
om distributies van data in een grafiekje op het scherm te zetten.

In [1]:
%load_ext autoreload
%autoreload 2

In [2]:
from spel import Spel

# Instellingen

In [3]:
N_SPELERS = 1024
START_KAPITAAL_SPELER = 100

WINST_FACTOR = 1.2
VERLIES_FACTOR = 0.8

def wereld():
    return Spel(N_SPELERS, START_KAPITAAL_SPELER, WINST_FACTOR, VERLIES_FACTOR)

# Spelen!

We laten eerst even zien dat onze spelwerelden reproduceerbaar zijn.

We starten de wereld.

In [4]:
W = wereld()

We gaan 1000 rondes spelen, en na elke 100 rondes kijken we hoe het kapitaal verdeeld is over de
bank en de spelers.

In [5]:
W.rondes(1000, 100, opnieuw=True)

Ronde   100: bank ~ spelers     109% ~      91%
Ronde   200: bank ~ spelers     142% ~      58%
Ronde   300: bank ~ spelers     156% ~      44%
Ronde   400: bank ~ spelers     167% ~      33%
Ronde   500: bank ~ spelers     187% ~      13%
Ronde   600: bank ~ spelers     197% ~       3%
Ronde   700: bank ~ spelers     197% ~       3%
Ronde   800: bank ~ spelers     197% ~       3%
Ronde   900: bank ~ spelers     197% ~       3%
Ronde  1000: bank ~ spelers     198% ~       2%


We doen een nieuw spel zonder de wereld opnieuw op te starten.

In [6]:
W.rondes(1000, 100, opnieuw=True)

Ronde   100: bank ~ spelers      67% ~     133%
Ronde   200: bank ~ spelers     107% ~      93%
Ronde   300: bank ~ spelers     108% ~      92%
Ronde   400: bank ~ spelers     150% ~      50%
Ronde   500: bank ~ spelers     193% ~       7%
Ronde   600: bank ~ spelers     178% ~      22%
Ronde   700: bank ~ spelers     194% ~       6%
Ronde   800: bank ~ spelers     200% ~       0%
Ronde   900: bank ~ spelers     200% ~       0%
Ronde  1000: bank ~ spelers     200% ~       0%


Het tweede spel heeft duidelijk een ander verloop.

## Reproduceer!

Nu starten we de wereld opnieuw.

In [7]:
W = wereld()

We spelen het eerste spel, en dit heeft precies het zelfde verloop
als het eerste spel in de vorige wereld.

In [8]:
W.rondes(1000, 100, opnieuw=True)

Ronde   100: bank ~ spelers     109% ~      91%
Ronde   200: bank ~ spelers     142% ~      58%
Ronde   300: bank ~ spelers     156% ~      44%
Ronde   400: bank ~ spelers     167% ~      33%
Ronde   500: bank ~ spelers     187% ~      13%
Ronde   600: bank ~ spelers     197% ~       3%
Ronde   700: bank ~ spelers     197% ~       3%
Ronde   800: bank ~ spelers     197% ~       3%
Ronde   900: bank ~ spelers     197% ~       3%
Ronde  1000: bank ~ spelers     198% ~       2%


Nu spelen we het tweede spel, dat precies hetzelfde verloop heeft als het tweede spel
in de vorige wereld.

In [9]:
W.rondes(1000, 100, opnieuw=True)

Ronde   100: bank ~ spelers      67% ~     133%
Ronde   200: bank ~ spelers     107% ~      93%
Ronde   300: bank ~ spelers     108% ~      92%
Ronde   400: bank ~ spelers     150% ~      50%
Ronde   500: bank ~ spelers     193% ~       7%
Ronde   600: bank ~ spelers     178% ~      22%
Ronde   700: bank ~ spelers     194% ~       6%
Ronde   800: bank ~ spelers     200% ~       0%
Ronde   900: bank ~ spelers     200% ~       0%
Ronde  1000: bank ~ spelers     200% ~       0%


# De eerste 10 zetten

Hierboven beweerden we dat bij 1000 spelers de eerste 10 zetten
nog niet een race-to-the-bottom vertonen, maar dat dat pas daarna optreedt.

Laten we in het eerste spelverloop hierboven eens gedetailleerd naar de eerste zetten kijken.

In [10]:
W = wereld()

In [11]:
W.rondes(10, 1, opnieuw=True, verdeling=True)

Ronde     1: bank ~ spelers      99% ~     101%
──────┬─────────────────────────────────────────────────────
bedrag│spelers
──────┼─────────────────────────────────────────────────────
   120│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■540
    80│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■484
──────┴─────────────────────────────────────────────────────
Ronde     2: bank ~ spelers      98% ~     102%
──────┬─────────────────────────────────────────────────────
bedrag│spelers
──────┼─────────────────────────────────────────────────────
   144│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■292
    96│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■489
    64│■■■■■■■■■■■■■■■■■■■■■■■■■243
──────┴─────────────────────────────────────────────────────
Ronde     3: bank ~ spelers      97% ~     103%
──────┬─────────────────────────────────────────────────────
bedrag│spelers
──────┼─────────────────────────────────────────────────────
   173│■■■■■■■■■■■■■■■■■■■■153
   115│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■

Merk op dat

* de verdeling tussen bank en spelers nagegenoeg gelijk gebleven is;
* de spelers zelf scheef-normaal verdeeld zijn geraakt over de bedragen.

Met scheef-normale verdeling bedoel ik:

Als je de verschillende bedragen die de spelers hebben beschouwt als emmers,
en je zet de emmers op een rij, en je stopt de spelers in de bijbehorende
emmers, dan zijn de spelers normaal verdeeld over de emmers.

Het scheve zit hem hierin, dat de rijkere emmers veel rijker zijn dan de armere
emmers arm zijn.

## De volgende 10 zetten

We zouden nu een overgang moeten merken.

In [12]:
W.rondes(10, 1, opnieuw=False, verdeling=True)

Ronde    11: bank ~ spelers      98% ~     102%
──────┬─────────────────────────────────────────────────────
bedrag│spelers
──────┼─────────────────────────────────────────────────────
   495│■■9
   330│■■■■■■27
   220│■■■■■■■■■■■■■■■■■81
   147│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■179
    98│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■230
    65│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■225
    43│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■146
    29│■■■■■■■■■■■■■■■■■■88
    19│■■■■■■■32
    13│■■6
     9│■1
──────┴─────────────────────────────────────────────────────
Ronde    12: bank ~ spelers      97% ~     103%
──────┬─────────────────────────────────────────────────────
bedrag│spelers
──────┼─────────────────────────────────────────────────────
   594│■■4
   396│■■■■■18
   264│■■■■■■■■■■■■■■■59
   176│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■135
   117│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■201
    78│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■222
    52│■■■■■■■■■■■■■■

De verdeling is nog steeds normaal, de bank en de spelers houden elkaar nog steeds in
evenwicht.

Wel groeien de spelers onderling steeds verder uitelkaar.

## Zetten 20 tot 30

In [13]:
W.rondes(10, 1, opnieuw=False, verdeling=True)

Ronde    21: bank ~ spelers      96% ~     104%
──────┬─────────────────────────────────────────────────────
bedrag│spelers
──────┼─────────────────────────────────────────────────────
  1363│■1
   909│■3
   606│■■■■12
   404│■■■■■■■■■■36
   269│■■■■■■■■■■■■■■■58
   180│■■■■■■■■■■■■■■■■■■■■■■■■■99
   120│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■141
    80│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■158
    53│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■180
    35│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■138
    24│■■■■■■■■■■■■■■■■■■■■■■■■■■100
    16│■■■■■■■■■■■■■■53
    11│■■■■■■■■30
     7│■■■9
     5│■■5
     3│■1
──────┴─────────────────────────────────────────────────────
Ronde    22: bank ~ spelers      95% ~     105%
──────┬─────────────────────────────────────────────────────
bedrag│spelers
──────┼─────────────────────────────────────────────────────
  1090│■■4
   727│■■■7
   485│■■■■■■■■■26
   323│■■■■■■■■■■■■■■■■45
   215│■■■■■■■■■■■■■■■■■■■■■■■■■■■■81
   144│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■

Nog steeds weinig verandering.

We kijken wat verder vooruit, en focussen op de verdeling bank/ spelers

In [14]:
W.rondes(100, 1, opnieuw=False, verdeling=False)

Ronde    31: bank ~ spelers      94% ~     106%
Ronde    32: bank ~ spelers      93% ~     107%
Ronde    33: bank ~ spelers      95% ~     105%
Ronde    34: bank ~ spelers      97% ~     103%
Ronde    35: bank ~ spelers      97% ~     103%
Ronde    36: bank ~ spelers      97% ~     103%
Ronde    37: bank ~ spelers      96% ~     104%
Ronde    38: bank ~ spelers      97% ~     103%
Ronde    39: bank ~ spelers      97% ~     103%
Ronde    40: bank ~ spelers      97% ~     103%
Ronde    41: bank ~ spelers      98% ~     102%
Ronde    42: bank ~ spelers     100% ~     100%
Ronde    43: bank ~ spelers     100% ~     100%
Ronde    44: bank ~ spelers     100% ~     100%
Ronde    45: bank ~ spelers     101% ~      99%
Ronde    46: bank ~ spelers     101% ~      99%
Ronde    47: bank ~ spelers     101% ~      99%
Ronde    48: bank ~ spelers     102% ~      98%
Ronde    49: bank ~ spelers     100% ~     100%
Ronde    50: bank ~ spelers      98% ~     102%
Ronde    51: bank ~ spelers      99% ~  

Pas in ronde 85 begint het tij voor langere tijd te keren.
Laten we terug gaan naar ronde 80 en vandaar weer gedetailleerd kijken.

In [15]:
W = wereld()
W.rondes(80, 10, opnieuw=False, verdeling=False)
W.rondes(10, 1, opnieuw=False, verdeling=True)

Ronde    10: bank ~ spelers      98% ~     102%
Ronde    20: bank ~ spelers      95% ~     105%
Ronde    30: bank ~ spelers      91% ~     109%
Ronde    40: bank ~ spelers      97% ~     103%
Ronde    50: bank ~ spelers      98% ~     102%
Ronde    60: bank ~ spelers      98% ~     102%
Ronde    70: bank ~ spelers      97% ~     103%
Ronde    80: bank ~ spelers     103% ~      97%
Ronde    81: bank ~ spelers     104% ~      96%
──────┬────────────────────────────────────────────────────
bedrag│spelers
──────┼────────────────────────────────────────────────────
  4563│■1
  3042│■■3
  2028│■■■4
  1352│■■■■■8
   901│■■■■■■11
   601│■■■■■■■13
   401│■■■■■■■■■16
   267│■■■■■■■■■■■■■■■■■■■■39
   178│■■■■■■■■■■■■■■■■■■■■■■42
   119│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■56
    79│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■63
    53│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■70
    35│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■85
    23│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■85
    16│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■

Wie zien hier inderdaad een soort overgang, heel geleidelijk,
naar een verdeling waarbij de spelers in een soort buik in de onderkant
van de distributie terechtkomen.

In het overzicht boven konden we zien dat tussen 110 en 120 er een dieptepunt
ontstaat voor de spelers.

We werken er naar toe en laten de rondes tussen 110 en 120 in detail zien:

In [16]:
W.rondes(20, 1, opnieuw=False, verdeling=False)
W.rondes(10, 1, opnieuw=False, verdeling=True)

Ronde    91: bank ~ spelers     104% ~      96%
Ronde    92: bank ~ spelers     104% ~      96%
Ronde    93: bank ~ spelers     106% ~      94%
Ronde    94: bank ~ spelers     106% ~      94%
Ronde    95: bank ~ spelers     105% ~      95%
Ronde    96: bank ~ spelers     105% ~      95%
Ronde    97: bank ~ spelers     105% ~      95%
Ronde    98: bank ~ spelers     109% ~      91%
Ronde    99: bank ~ spelers     109% ~      91%
Ronde   100: bank ~ spelers     109% ~      91%
Ronde   101: bank ~ spelers     110% ~      90%
Ronde   102: bank ~ spelers     111% ~      89%
Ronde   103: bank ~ spelers     106% ~      94%
Ronde   104: bank ~ spelers     107% ~      93%
Ronde   105: bank ~ spelers     110% ~      90%
Ronde   106: bank ~ spelers     106% ~      94%
Ronde   107: bank ~ spelers     106% ~      94%
Ronde   108: bank ~ spelers     109% ~      91%
Ronde   109: bank ~ spelers     112% ~      88%
Ronde   110: bank ~ spelers     115% ~      85%
Ronde   111: bank ~ spelers     115% ~  

De buik is gegroeid.
Maar de rijkste speler is ook behoorlijk rijk geworden.
Die biedt weerstand tegen de bank.
Dat zien we in de komende 10 zetten gebeuren:

In [17]:
W.rondes(10, 1, opnieuw=False, verdeling=True)

Ronde   121: bank ~ spelers     113% ~      87%
──────┬─────────────────────────────────────────────────────
bedrag│spelers
──────┼─────────────────────────────────────────────────────
 10210│■1
  6806│■2
  4538│■1
  2017│■2
  1344│■■4
   896│■■■8
   598│■■■■9
   398│■■■■■■■18
   266│■■■■■■■■■■■■34
   177│■■■■■■■■■■29
   118│■■■■■■■■■■■■■■41
    79│■■■■■■■■■■■■■■■■47
    52│■■■■■■■■■■■■■■■■■■■54
    35│■■■■■■■■■■■■■■■■■■51
    23│■■■■■■■■■■■■■■■■■■■■■■63
    16│■■■■■■■■■■■■■■■■■48
    10│■■■■■■■■■■■■■■■■■■■■■■■■■■76
     7│■■■■■■■■■■■■■■■■■■■■■■■67
     5│■■■■■■■■■■■■■■■■■■■■■62
     3│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■84
     2│■■■■■■■■■■■■■■■■■■■■■■■■70
     1│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■144
     0│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■109
──────┴─────────────────────────────────────────────────────
Ronde   122: bank ~ spelers     111% ~      89%
──────┬─────────────────────────────────────────────────────
bedrag│spelers
──────┼─────────────────────────────────────────

De spelers zijn nu weer behoorlijk ingelopen op de bank.

Maar let wel, dat is niet zozeer een collectieve actie van de spelers. 
Het zijn de rijkste spelers die zich nog verder verrijkt hebben.

## Kapitalistische dynamiek

De tendens is dat al het kapitaal verhuist van de spelers naar de bank.
Maar er kunnen behoorlijke fluctuaties optreden.
Ook in ver gevorderde stadia kan er toch plots veel kapitaal van de bank
naar de spelers vloeien.

Hoe gaat dat precies in zijn werk? Welke dynamiek zien we?

Laten we de wereld starten en tien spelen van duizend rondes laten lopen,
waarna we gaan kijken hoe die spelverlopen eruitzien.

In [18]:
W = wereld()

for s in range(10):
    print("-" * 50)
    print(f"Spel {s + 1:>2}")
    W.rondes(1000, 100, opnieuw=True)

--------------------------------------------------
Spel  1
Ronde   100: bank ~ spelers     109% ~      91%
Ronde   200: bank ~ spelers     142% ~      58%
Ronde   300: bank ~ spelers     156% ~      44%
Ronde   400: bank ~ spelers     167% ~      33%
Ronde   500: bank ~ spelers     187% ~      13%
Ronde   600: bank ~ spelers     197% ~       3%
Ronde   700: bank ~ spelers     197% ~       3%
Ronde   800: bank ~ spelers     197% ~       3%
Ronde   900: bank ~ spelers     197% ~       3%
Ronde  1000: bank ~ spelers     198% ~       2%
--------------------------------------------------
Spel  2
Ronde   100: bank ~ spelers      67% ~     133%
Ronde   200: bank ~ spelers     107% ~      93%
Ronde   300: bank ~ spelers     108% ~      92%
Ronde   400: bank ~ spelers     150% ~      50%
Ronde   500: bank ~ spelers     193% ~       7%
Ronde   600: bank ~ spelers     178% ~      22%
Ronde   700: bank ~ spelers     194% ~       6%
Ronde   800: bank ~ spelers     200% ~       0%
Ronde   900: bank 

Merk de volgende dingen op

* In vergevordere stadia van het spel kunnen spelers nog kapitaal van de bank
  terugpakken.
* Spelers kunnen samen aanzienlijk meer veroveren dan wat ze gezamenlijk in het begin
  hadden.
* Er zijn extreme pieken bij, waar de spelers een vermogen veroveren dat
  wel 7,5 keer zo groot is als wat de bank ooit had (spel 7). De bank staat flink rood.
* Die pieken altijd gevolgd worden door evenzovele dalingen: de spelers
  kunnen het kapitaal niet vasthouden.
* Uiteindelijk wint de bank het altijd.

We zagen al eerder dat in een situatie van terugpakken van kapitaal
door de spelers dat gebeurt door de rijkste spelers.
Wat opvalt is dat na hoge pieken in het spelerskapitaal, het spelerskapitaal weer
heel snel zakt.

We focussen op spel 3, vanaf ronde 400.
We herscheppen die situatie.

In [22]:
W = wereld()

for s in range(2):
    W.rondes(1000, 1000, opnieuw=True)
    
W.rondes(400, 100, opnieuw=True)

Ronde  1000: bank ~ spelers     198% ~       2%
Ronde  1000: bank ~ spelers     200% ~       0%
Ronde   100: bank ~ spelers     117% ~      83%
Ronde   200: bank ~ spelers     109% ~      91%
Ronde   300: bank ~ spelers     104% ~      96%
Ronde   400: bank ~ spelers      52% ~     148%


Nu zijn we precies bij die 148% voor de spelers, waarna het snel naar de 52% zal gaan.

Laten we de verdeling van kapitaal over de spelers eens bekijken.

In [23]:
W.toon()

Ronde   400: bank ~ spelers      52% ~     148%
──────┬─────────────────────────────────────────────────────
bedrag│spelers
──────┼─────────────────────────────────────────────────────
139867│■1
  3638│■1
  1617│■1
  1078│■1
   479│■2
   319│■3
   213│■2
   142│■5
    95│■6
    63│■6
    42│■5
    28│■14
    19│■8
    12│■■17
     8│■14
     6│■■17
     4│■■19
     2│■■■40
     1│■■■■■67
     0│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■795
──────┴─────────────────────────────────────────────────────


Aha!
Er is een supperrijke speler ontstaan samen met een paar zeer rijken.
De superrijke is wel 40x zo rijk als de tweede rijke.
Het gros van de spelers is al failliet.

De bank speelt dus nog maar met weinig spelers, en dat is in het voordeel van de bank.
Want zolang er veel spelers zijn, treden er nog veel verschillende spelverlopen op,
en wordt de bank geconfronteerd met sterke winnaars.
Maar met weinig spelers zal de bank al gauw met elk van die spelers in een spelverloop
terechtkomen waar de speler even vaak wint als verliest.
En dat is voor die speler een race-to-the-bottom.

Kortom, het lijkt even wat minder goed te gaan voor de bank op dit moment.
Maar de extreme ongelijkheid van de spelers onderling zorgt ervoor dat de bank zich
geen zorgen hoeft te maken.
Ze gaat vanaf nu het geld terugpakken van de spelers.

We gaan de volgende 100 rondes nalopen, en na elke 10 rondes laten we de verdeling zien.

In [24]:
W.rondes(100, 10, opnieuw=False, verdeling=True)

Ronde   410: bank ~ spelers     141% ~      59%
──────┬─────────────────────────────────────────────────────
bedrag│spelers
──────┼─────────────────────────────────────────────────────
 50686│■1
  2967│■1
  1318│■1
   586│■3
   391│■1
   260│■4
   174│■1
   116│■5
    77│■6
    51│■8
    34│■7
    23│■3
    15│■15
    10│■■18
     7│■12
     5│■■17
     3│■■23
     2│■■21
     1│■■■■■■83
     0│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■794
──────┴─────────────────────────────────────────────────────
Ronde   420: bank ~ spelers      97% ~     103%
──────┬─────────────────────────────────────────────────────
bedrag│spelers
──────┼─────────────────────────────────────────────────────
 92988│■1
  3628│■1
  2419│■1
  1075│■1
   717│■2
   319│■1
   212│■4
   142│■2
    94│■8
    63│■3
    42│■6
    28│■7
    19│■13
    12│■■17
     8│■12
     6│■10
     4│■■19
     2│■■■42
     1│■■■■60
     0│■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■814
──────┴─────────────────────────────

Al in 10 rondes heeft de bank de rollen omgekeerd.
Maar de spelers geven zich nog niet gewonnen, ze slagen er toch in om weer terug te komen
op het collectieve oorspronkelijke spelerskapitaal.
Ze raken dat toch snel weer kwijt, ze winnen het nog een keer terug, maar dan gaat het definitief
bergafwaarts.
Al die tijd raken er meer spelers failliet, zodat gezamenlijke kracht van de spelers
verder ondermijnd wordt.

# Conclusies

## Er is meer dan asymptotisch gedrag

Dit YardSale-achtige spelletje van een bank met zijn klanten laat een verrassend gevarieerde
dynamiek zien, die niet gevangen is in een simpele wetmatigheid.

Het is wel zo, dat de systeemparameters er voor zorgen dat uiteindelijk de bank altijd wint.
Maar voor het zo ver is, vindt er een soort titanenstrijd plaats tussen de bank en
sommige (zeer) rijk geworden klanten, waarbij de kansen verschillende malen keren.

Het tragische is dat de spelers alleen kunnen winnen door zich collectief te verzwakken,
en de strijd uit te besteden aan de superrijken.
Die kunnen dan eventjes winnen, maar de verzwakking van het collectief eist daarna zijn tol,
en alle klanten gaan uiteindelijk failliet.

## Simpele simulaties kunnen inzicht verschaffen

Als zulk een eenvoudig model zoveel herkenbare dynamiek laat zien, dan is het voor mij
een aanwijzing dat het maken en doorrekenen van dit soort modellen een goede methode
kan zijn om de macro-effecten van eerlijk-lijkende transactie parameters te ontdekken.

Lees de referenties bij het artikel in de Scientific American waar we mee begonnen, en je ziet
dat dat inderdaad volop gebeurt.

## Hou het zo simpel mogelijk

In de discussie op KlimaatVeranda verzette Hans zich tegen methoden die de zaken
nodeloos gecompliceerd maken.

Naar mijn mening moet dat niet doorschieten naar het gebruik van methoden die de
complexiteit van het probleem niet aankunnen.

En, het is misschien wel wennen dat een model met zo weinig parameters iets kan zeggen
over een werkelijkheid die zo ingewikkeld is als de economische. 
Op die manier beschouwd is het Yard Sale model een prima gebruik van
[Occams scheermes](https://nl.wikipedia.org/wiki/Ockhams_scheermes).